section-alert: add low progress tones (#1996)
agriculturegovau/agds-next: packages/react/src/section-alert/docs/code.mdx

## Props

<ComponentPropsTable name="SectionAlert" />

> ## Focusing the alert

Reviewer: Why is this moving to Code?

Author: It wasn't in the design docs, so I moved it to `code`. Moved it back to the main page.

_NOTE: When focusing an alert after closing a `Drawer`, use the [focusElementOnClose prop in the Drawer component](/components/drawer#focus-custom-element-on-close)._

In the event you need to focus the alert, you can do so in one of three ways:

1. [Automatically focus on mount](#automatically-focus-on-mount).
2. [Automatically focus on update](#automatically-focus-on-update).
3. [Manually set focus](#manually-set-focus).

### Automatically focus on mount

> Press the "Toggle alert" button below to mount/unmount the alert. Notice that each time the alert is mounted it will be focused. Here we use the `focusOnMount` boolean prop to automatically handle the focusing.

Reviewer: ```suggestion
Press the “Toggle alert” button below to mount/unmount the alert. Notice that each time the alert is mounted it will be focused. Here we use the `focusOnMount` boolean prop to automatically handle the focus management.
```

```jsx live
() => {
	const [showAlert, setShowAlert] = React.useState(false);

	return (
		<Stack gap={2}>
			<Button
				alignSelf="start"
				onClick={() => setShowAlert((prevShowAlert) => !prevShowAlert)}
			>
				Toggle alert
			</Button>

			{showAlert && (
				<SectionAlert
					focusOnMount
					title="Something went wrong"
					tone="errorHigh"
				>
					<Text as="p">There was an error submitting the application.</Text>
				</SectionAlert>
			)}
		</Stack>
	);
};
```

### Automatically focus on update

> Press the "Toggle alert" button below to mount/unmount the alert. Notice that each time the alert is mounted and subsequently updated it will be focused. Here we use the `focusOnUpdate` prop to automatically handle the focusing both on mount, and on updates to the alert's content.

Author: Updated in [d649ec](https://github.com/agriculturegovau/agds-next/pull/1996/commits/d649ec1f3d96f8ef4c300512d614545bf689ece5). Moved back into `overview.mdx`

> Once the alert is displayed, try updating the content by pressing the "Update alert content" button. This will focus the alert.

Reviewer: ```suggestion
Once the alert is displayed, update the content by pressing the “Update alert content” button. Focus will be automatically moved to the alert.
```

Author: Updated in [d649ec](https://github.com/agriculturegovau/agds-next/pull/1996/commits/d649ec1f3d96f8ef4c300512d614545bf689ece5). Moved back into `overview.mdx`

```jsx live
() => {
	const availableErrors = [
		'Full name must not be empty.',
		'Email must not be empty.',
	];

	const [errors, setErrors] = React.useState([]);

	return (
		<Stack gap={2}>
			<Button
				alignSelf="start"
				onClick={() =>
					// Here we mock the updating of errors in our list
					setErrors((prevErrors) =>
						availableErrors
							.toReversed()
							.slice(
								availableErrors.length - prevErrors.length - 1,
								availableErrors.length - prevErrors.length + 1
							)
							.reverse()
					)
				}
			>
				Update alert content
			</Button>

			{!!errors.length && (
				<SectionAlert
					focusOnUpdate={errors}
					title="Something went wrong"
					tone="errorHigh"
				>
					<UnorderedList>
						{errors.map((text) => (
							<ListItem key={text}>
								<Text as="p">{text}</Text>
							</ListItem>
						))}
					</UnorderedList>
				</SectionAlert>
			)}
		</Stack>
	);
};
```

### Manually set focus

> Press the "Focus the alert" button below to set focus on the alert. To achieve this, we set the `tabIndex` to `-1`. This will allow programmatic focusing and will render a focus ring to aid accessibility.

Reviewer: ```suggestion
Press the “Focus the alert” button below to manually set focus on the alert. To achieve this, we must set `tabIndex` to `-1` to allow programmatic focusing.
```

Author: Updated in [d649ec](https://github.com/agriculturegovau/agds-next/pull/1996/commits/d649ec1f3d96f8ef4c300512d614545bf689ece5). Moved back into `overview.mdx`
